src/script/standard.h

```diff
@@ -80,9 +80,15 @@ struct PKHash : public uint160
     using uint160::uint160;
 };
 
+struct WitnessV0KeyHash;
+struct WitnessV0ScriptHash;
 struct ScriptHash : public uint160
 {
     ScriptHash() : uint160() {}
+    // These don't do what you'd expect.
+    // Use ScriptHash(GetScriptForDestination(...)) instead.
+    explicit ScriptHash(const WitnessV0KeyHash& hash) = delete;
+    explicit ScriptHash(const WitnessV0ScriptHash& hash) = delete;
```

Comment:
Instead of deleting them, couldn't you just call `GetScriptForDestination` in the constructor?

Reply:
Yes, but that means I have to think more if we want to support more conversions, like `ScriptHash(PKHash())` etc. 

Like, is this a general tool for generating output scripts, or what?

Reply:
Yeah, I guess highest priority is fixing the sharp edge. One way or another.

Reply:
> Instead of deleting them, couldn't you just call GetScriptForDestination in the constructor?

 Sounds like begging for sharp edges in backporting...

Reply:
Thank you